content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,235 @@
+hardcode the address of the account, and save the complexities of key handling for later. The application, after finishing 
+this step should look like this:
+
+![Screenshot: Step 3, show account information](img/javascript-wallet-3.png)
+
+To get the application running at this stage of development, run the following command:
+
+```console
+npm run account
+```
+
+### 4. Show Account's Transactions
+
+**Full code for this step:** [`4_tx-history.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/4_tx-history.js).
+
+At this point, your wallet shows the account's balance getting updated, but doesn't give you any clue about how this state
+came about, namely the actual transactions that caused the updates. So, our next step is to display the account's 
+transaction history and to keep it up to date:
+
+To get the application running at this stage of development, run the following command:
+
+![Screenshot: Step 4, show transaction history](img/javascript-wallet-4.png)
+
+```console
+npm run tx-history
+```
+
+
+### 5. Send XRP
+
+**Full code for this step:** [`5_send-xrp.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/2_send-xrp.js).
+
+Up until now we have enabled our app to query and display data from the XRPL. Now it's time to actively participate in
+the ledger by enabling out application to send transactions. For now, you can stick to sending direct XRP payments 
+because there are more complexities involved in sending issued tokens. 
+
+But before that, it is time to take that wallflowery appearance of our application and give it a more usable and 
+nicer looking UI. 
+
+![Screenshot: Step 5, send xrp dialog](img/javascript-wallet-5.png)
+
+To get the application running at this stage of development, run the following command:
+
+```console
+npm run send-xrp
+```
+
+
+### 6. Domain Verification and Polish
+
+**Full code for this step:** [`6_domain-verification.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/3_domain-verification.js).
+
+One of the biggest shortcomings of the wallet app from the previous step is that it doesn't provide a lot of 
+protections or feedback for users to save them from human error and scams. These sorts of protections are extra 
+important when dealing with the cryptocurrency space, because decentralized systems like the XRP Ledger don't 
+have an admin or support team you can ask to cancel or refund a payment if you made a mistake such as sending it to 
+the wrong address. This step shows how to add some checks on destination addresses to warn the user before sending.
+
+One type of check you can make is to verify the domain name associated with an XRP Ledger address; this is called
+[account domain verification](xrp-ledger-toml.html#account-verification). When an account's domain is verified, 
+you could show it like this:
+
+TBD
```

Comment:
Just marking so one of the TBD shows up while you're reviewing comments - I know you said you would add detail once reviewed.